bootstrap/src/main/java/org/jboss/windup/bootstrap/commands/windup/ToolingModeCommand.java

```diff
@@ -0,0 +1,198 @@
+        return arguments.get(addDirectoryIndex);
+    }
+
+    @Override
+    public CommandResult execute()
+    {
+        try
+        {
+            furnace = FurnaceFactory.getInstance();
+            furnace.setServerMode(true);
+            loadAddons();
+            try
+            {
+                Future<Furnace> future = furnace.startAsync();
+                future.get();
+            }
+            catch (Exception e)
+            {
+                System.out.println("Failed to start "+Util.WINDUP_BRAND_NAME_ACRONYM+"!");
+                e.printStackTrace();
+            }
+            this.analyze();
+            this.furnace.stop();
+        }
+        catch (Throwable t)
+        {
+            System.err.println(Util.WINDUP_BRAND_NAME_ACRONYM + " execution failed due to: " + t.getMessage());
+            t.printStackTrace();
+        }
+        return null;
+    }
+
+    private void loadAddons()
+    {
+        AddImmutableAddonDirectoryCommand addonCommand = new AddImmutableAddonDirectoryCommand(addonsDirectory);
+        addonCommand.setFurnace(furnace);
+        addonCommand.execute();
+    }
+
+    private void analyze()
+    {
+        System.out.println("Calling ToolingModeRunner...");
+        Set<String> input = this.getInput();
+        String output = this.getOutput();
+        boolean sourceMode = this.isSourceMode();
+        boolean ignoreReport = this.ignoreReport();
+        List<String> ignorePatterns = this.getIgnorePatterns();
+        String windupHome = this.getWindupHome();
+        List<String> source = this.getSource();
+        List<String> target = this.getTarget();
+        List<File> rulesDir = this.getUserRulesDir();
+        System.out.println("Using Data: ");
+        System.out.println("input: " + input);
+        System.out.println("output: " + output);
+        System.out.println("sourceMode: " + sourceMode);
+        System.out.println("skipReport: " + ignoreReport);
+        System.out.println("ignorePatterns: " + ignorePatterns);
+        System.out.println("windupHome: " + windupHome);
+        System.out.println("source: " + source);
+        System.out.println("target: " + target);
+        System.out.println("userRulesDirectory: " + rulesDir);
+        furnace.getAddonRegistry().getServices(ToolingModeRunner.class).get()
+            .run(input, output, sourceMode, ignoreReport, ignorePatterns,
+                    windupHome, source, target, rulesDir);
+    }
+
+    @Override
+    public CommandPhase getPhase()
+    {
+        return null;
+    }
+
+    public static boolean isToolingMode(List<String> arguments)
+    {
+        return arguments.contains(ToolingModeCommand.COMMAND_ID);
+    }
+
+    private Set<String> getInput()
+    {
+        int index = arguments.indexOf(toArg(IOptionKeys.INPUT)) + 1;
+        return Sets.toSet(this.getValues(index));
+    }
+
+    private String getOutput()
+    {
+        int outputIndex = arguments.indexOf(toArg(IOptionKeys.OUTPUT)) + 1;
```

Comment:
if the argument expected doesnt exist the code will take another argument as the value for the Input argument.
Should we check and complain or default ?

Reply:
Argument check has been added.